### PR TITLE
Closes #1735: Rename all "WebView" that are actually "EngineView"

### LINKED ISSUE #1735: Rename confusing "webview" to differentiate between EngineView and WebView
As part of #1720 it's confusing that both EngineView and WebView use the variable `webView`
https://github.com/mozilla-mobile/firefox-tv/pull/1726/commits/10f59c7c16af6120d3c08257291fd63211a8d46e#r251492765

This should be very small to rename, with an improvement in clarity (because the Engine/Session/WebView architecture from AC is already fairly complex).

### PR DESCRIPTION
**Pull request by psymoon**
### Pull Request checklist

- [x] This PR includes thorough **tests** or an explanation of why it does not
- [x] This PR includes a **CHANGELOG entry** or does not need one
- [x] The **UI tests** are passing after this PR (`./gradlew connectedDebugAndroidTest`)
- [x] I have considered adding **QA labels** on the associated issue (not this PR; `qa-ready` or `qa-denied`)

### CODE REVIEW

File: app/src/main/java/org/mozilla/tv/firefox/ext/EngineView.kt [hunk 1]
```diff
@@ -29,7 +29,7 @@ fun EngineView.setupForApp() {
     // Also increase text size to fill the viewport (this mirrors the behaviour of Firefox,
     // Chrome does this in the current Chrome Dev, but not Chrome release).
     // TODO #33: TEXT_AUTOSIZING does not exist in AmazonWebSettings
-    // webView.settings.setLayoutAlgorithm(AmazonWebSettings.LayoutAlgorithm.TEXT_AUTOSIZING);
+    // engineView.settings.setLayoutAlgorithm(AmazonWebSettings.LayoutAlgorithm.TEXT_AUTOSIZING);
```
Comment (liuche) [inline app/src/main/java/org/mozilla/tv/firefox/ext/EngineView.kt]: Should this still be webView?

File: app/src/main/java/org/mozilla/tv/firefox/ext/EngineView.kt [hunk 2]
```diff
@@ -155,7 +155,7 @@ private class EngineViewExtension(private val engineView: EngineView) {
                 (sessionManager.getOrCreateEngineSession() as SystemEngineSession).webView
             } else {
                 // After clearing all session we temporarily don't have a selected session
-                // and [SessionRepo.clear()] destroyed the existing webView - see [SystemEngineView.onDestroy()]
+                // and [SessionRepo.clear()] destroyed the existing engineView - see [SystemEngineView.onDestroy()]
```
Comment (liuche) [inline app/src/main/java/org/mozilla/tv/firefox/ext/EngineView.kt]: Also still webView?

File: app/src/main/java/org/mozilla/tv/firefox/webrender/EngineViewLifecycleFragment.kt [hunk 3]
```diff
@@ -34,44 +34,44 @@ abstract class EngineViewLifecycleFragment : LocaleAwareFragment() {
...
+    open fun onEngineViewCreated(engineView: EngineView) = Unit
 
     override fun onStop() {
         super.onStop()
 
         // NB: onStop unexpectedly calls onPause: see below.
         //
         // When the user says "Alexa pause [the video]", the Activity will be paused/resumed while
-        // Alexa handles the request. If the WebView is paused during video playback, the video will
+        // Alexa handles the request. If the EngineView is paused during video playback, the video will
```
Comment (liuche) [inline app/src/main/java/org/mozilla/tv/firefox/webrender/EngineViewLifecycleFragment.kt]: Do we have a concept of EngineView being paused? I could see this being "webView inside EngineView", but just wanted to clarify.

File: app/src/main/java/org/mozilla/tv/firefox/webrender/EngineViewLifecycleFragment.kt [hunk 4]
```diff
@@ -34,44 +34,44 @@ abstract class EngineViewLifecycleFragment : LocaleAwareFragment() {
...
-        // to not call WebView.onPause so we pause the WebView later, here in onStop, when it isn't
+        // to not call EngineView.onPause so we pause the EngineView later, here in onStop, when it isn't
         // affected by Alexa voice commands. Luckily, Alexa pauses the video for us. afaict, on
         // Fire TV, `onPause` without `onStop` isn't called very often so I don't think there will
         // be many side effects (#936).
         //
         // The problem is not reproducible when onPause is called here in onStop (even if pauseTimers is
         // called in onPause, in the android-components library).
-        webView!!.onPauseIfNotNull() // internally calls WebView.onPause: see impl for details.
+        engineView!!.onPauseIfNotNull() // internally calls EngineView.onPause: see impl for details.
```
Comment (liuche) [inline app/src/main/java/org/mozilla/tv/firefox/webrender/EngineViewLifecycleFragment.kt]: "internally calls WebView.onPause"

File: app/src/main/java/org/mozilla/tv/firefox/webrender/VideoVoiceCommandMediaSession.kt [hunk 5]
```diff
@@ -171,7 +171,7 @@ class VideoVoiceCommandMediaSession @UiThread constructor(
         //
         // The videos may send playback state update events to Java, which we're forced to ignore:
         // see JavascriptVideoPlaybackStateSyncer for the code.
-        webView?.evalJS("document.querySelectorAll('video').forEach(v => v.pause());")
+        engineView?.evalJS("document.querySelectorAll('video').forEach(v => v.pause());")
```
Comment (liuche) [inline app/src/main/java/org/mozilla/tv/firefox/webrender/VideoVoiceCommandMediaSession.kt]: huh, I didn't realize that but I guess the engineView can call evalJS.

File: app/src/main/java/org/mozilla/tv/firefox/ext/EngineView.kt (hunk 1 above)
Comment (psymoon) [reply app/src/main/java/org/mozilla/tv/firefox/ext/EngineView.kt]: Yes

File: app/src/main/java/org/mozilla/tv/firefox/ext/EngineView.kt (hunk 2 above)
Comment (psymoon) [reply app/src/main/java/org/mozilla/tv/firefox/ext/EngineView.kt]: Yes

File: app/src/main/java/org/mozilla/tv/firefox/webrender/EngineViewLifecycleFragment.kt (hunk 3 above)
Comment (psymoon) [reply app/src/main/java/org/mozilla/tv/firefox/webrender/EngineViewLifecycleFragment.kt]: We do have `EngineView.onPause()`

File: app/src/main/java/org/mozilla/tv/firefox/webrender/EngineViewLifecycleFragment.kt (hunk 4 above)
Comment (psymoon) [reply app/src/main/java/org/mozilla/tv/firefox/webrender/EngineViewLifecycleFragment.kt]: That's correct - It is `EngineView.onPause()`

File: app/src/main/java/org/mozilla/tv/firefox/webrender/VideoVoiceCommandMediaSession.kt (hunk 5 above)
Comment (psymoon) [reply app/src/main/java/org/mozilla/tv/firefox/webrender/VideoVoiceCommandMediaSession.kt]: Well it's an Engine ext method FWIW

**Pull request merged**